rewrite-java/src/main/java/org/openrewrite/java/internal/template/Substitutions.java

```diff
@@ -138,12 +139,19 @@ public String substitute() {
     }
 
     private String getTypeName(@Nullable JavaType type) {
-        if (type == null) {
-            return "java.lang.Object";
+        if (type instanceof JavaType.Parameterized) {
+            return ((JavaType.Parameterized) type).getFullyQualifiedName()
+                    + '<'
+                    + ((JavaType.Parameterized) type).getTypeParameters().stream().map(this::getTypeName).collect(Collectors.joining(","))
```

Comment:
please avoid streams in places like this

Reply:
I will refactor it. Possibly this could be a useful method for `TypeUtils`.